### PR TITLE
fix a major bug in ba.rand32f

### PR DESCRIPTION
**Pull request by Goober5000**
The function should return the random result, not the maximum value.

Also use `ade_set_args` for consistency.

### CODE REVIEW

**Review by jg18 — APPROVED** [review]
I don't know a ton about the scripting code, but this looks reasonable.

I wonder what the difference is between the current `ade_set_error()` and `ade_set_args()`?

**Comment by jg18** [comment]
Btw, despite the `rand32f` scripting function's documentation, there is currently some extremely small chance that `result` could be `1.0f`, if `Random::next()` just happens to return `Random::MAX_VALUE`.

I guess we could fix this by storing the value returned by `Random::next()` and decrementing it if it happens to be equal to `Random::MAX_VALUE`.

**Comment by Goober5000** [comment]
> I wonder what the difference is between the current `ade_set_error()` and `ade_set_args()`?

There is currently no difference.  But I think the original intent, and possibly an aspiration for the future, would be for `ade_set_error` to set an error flag which would be recognized by Lua or by the game.


> Btw, despite the `rand32f` scripting function's documentation, there is currently some extremely small chance that `result` could be `1.0f`, if `Random::next()` just happens to return `Random::MAX_VALUE`.

After quite a bit of searching online, it looks like this is ok; and I've also confirmed from the C standard that the random range is inclusive of `RAND_MAX`.  I've updated the documentation.

File: code/scripting/api/libs/base.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ ADE_FUNC(rand32f,
 	if (numargs > 0)
 		result *= _max;
```
[the remark this answers is not in the document]
Comment (Goober5000) [reply code/scripting/api/libs/base.cpp:99]: Possibly, but it doesn't break anything

File: code/scripting/api/libs/base.cpp [hunk 2]
```diff
@@ -93,12 +93,17 @@ ADE_FUNC(rand32f,
 	float _max;
 	int numargs = ade_get_args(L, "|f", &_max);
 
-	float result = (float)Random::next() * Random::INV_F_MAX_VALUE;
+	// see also frand()
+	int num;
+	do {
+		num = Random::next();
+	} while (num == Random::MAX_VALUE);
```
Comment (jg18) [inline code/scripting/api/libs/base.cpp:100]: Are we sure there aren't values close to `MAX_AVLUE` like `MAX_VALUE - 1` that when multiplied by `INV_F_MAX_VALUE` will result in `1.0f` due to `float` imprecision?

If there are, we should exclude those values aw well.

**Pull request merged**